cmd/history-exporter/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/google/go-github/v65/github"
+	"github.com/neondatabase/gh-workflow-stats-action/pkg/config"
+	"github.com/neondatabase/gh-workflow-stats-action/pkg/db"
+	"github.com/neondatabase/gh-workflow-stats-action/pkg/export"
+	"github.com/neondatabase/gh-workflow-stats-action/pkg/gh"
+)
+
+func main() {
+	var startDateStr string
+	var endDateStr string
+	var startDate time.Time
+	var endDate time.Time
+
+	flag.StringVar(&startDateStr, "start-date", "", "start date to quert and export")
+	flag.StringVar(&endDateStr, "end-date", "", "end date to quert and export")
+	flag.Parse()
+
+	if startDateStr == "" {
+		startDate = time.Now().Truncate(24 * time.Hour)
+	} else {
+		var err error
+		startDate, err = time.Parse("2006-01-02", startDateStr)
```

Comment:
Just curious, why `2006-01-02`?

Reply:
it's "Format" string to parse the date in Golang, and I also often confused about it, especially when working with other languages.
so here "2006" encodes "parse year with 4 digits" (analog of "%Y" in other languages), "01" means "2 digits month" ("%m" analog) and "02" means "2 digits day" ("%d" analog)
https://github.com/golang/go/blob/master/src/time/format.go#L152 with full format definitions